src/Pages/components/ContactList.jsx

```diff
@@ -0,0 +1,52 @@
+import { Link } from "react-router-dom";
+import { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { getRequest } from "../../Api";
+
+export default function ContactList({ limit }) {
+  const [contacts, setContacts] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const runEffect = async () => {
+      const { data, error } = await getRequest("/contact");
+      if (data) {
+        let finalisedContacts = data;
+        if (limit) {
+          finalisedContacts = data.filter((contact, index) => index < limit);
+        }
+        setContacts(finalisedContacts);
+      } else {
+        console.error("Error fetching contacts:", error);
+      }
+      setLoading(false);
+    };
+    runEffect();
+  }, [limit]);
+
+  return (
+    <>
+      {loading && <p>Loading...</p>}
+      {!loading && (
+        <div>
+          {contacts.map((contact) => {
```

Comment:
If there are no contacts to display, it might be useful for user experience to show a message indicating that there are no contacts available:
`{!loading && contacts.length === 0 && <p>No contacts available.</p>}
`